### DISCUSSION TITLE
Resize video clip on the fly for Telegram messaging

### DISCUSSION THREAD

**Discussion by ediguidedog** [blakeblackshear/frigate#5211] [Ask A Question] [answered]
Hi Is there a way to resize a video clip to something much smaller before I send it to Telegram? Currently I'm passing  
url: https://frigate.mydomain.com/api/events/{{trigger.payload_json["after"]["id"]}}/clip.mp4 which sends the full size clip which can take some time to be sent and delivered so I was wondering if there is a way I can convert it on the fly and send something low res that is suitable for a phone screen.

I know it can be done for stills using the existing api calls but anyone know of a way it can be done for video. 

Thanks

**Comment by NickM-27** [comment]
This is not currently supported

**Reply by ediguidedog** [reply]
Thanks for the super quick reply. Any idea if there is a way to do this in homeassistant by calling ffmpeg and passing the above URL as part of the automation?

**Reply by NickM-27** [reply]
yes it can definitely be done using ffmpeg, not sure how to inside HA as I haven't done that

**Comment by ediguidedog** [comment, marked as answer]
Just updating based on my fiddling. What I required was a (very) lowres copy of the event video for sending to telegram as when I'm out and about downloading megabytes of video is painfully show. 320x240 will suffice

I've managed this is a somewhat messy way using home assistant (running in a docker) that I was already using to manage notifications 

Firstly I had to add the following to the home assistant configuration.yaml

```
ffmpeg:  # enables ffmpeg support in HA

homeassistant:
  allowlist_external_dirs:  # allows access to folders in container
    - '/config'

shell_command:
  remuxvideo: ffmpeg -y -i "{{infile}}" -vf scale=320:240 "{{outfile}}".mp4   # this is the command that does the scaling 

```

Then I created an automation. The triggers and conditions are as normal for any MQTT event from frigate but the actions are as below 

```
action:
  - delay:
      hours: 0
      minutes: 0
      seconds: 2
      milliseconds: 0
  - service: shell_command.remuxvideo
    data:
      infile: >-
        https://myfrigateinstallation.com/api/events/{{trigger.payload_json["after"]["id"]}}/clip.mp4
      outfile: "{{trigger.payload_json[\"after\"][\"camera\"]}}"
  - delay:
      hours: 0
      minutes: 0
      seconds: 1
      milliseconds: 0
    enabled: true
  - service: telegram_bot.send_video
    data:
      target: 1234567890  # my telegram id/group
      timeout: 30
      file: /config/{{trigger.payload_json["after"]["camera"]}}.mp4
      parse_mode: html
mode: single
trace:
  stored_traces: 50
```

So this drops a 320x200 video file in the /config directory of the HA docker container with the filename of the camera that generated the event. The same video is then sent to telegram in the next step. The file is simple overwritten each time 

It works. The two delays just seem to be needed to let the writes complete as very occasionally without them it fails or I get a blank video.

Hopefully someone will find this useful. If anyone knows of a cleaner way shout up.
